nsqd/options.go

```diff
@@ -104,7 +136,7 @@ func NewOptions() *Options {
 		TCPAddress:        "0.0.0.0:4150",
 		HTTPAddress:       "0.0.0.0:4151",
 		HTTPSAddress:      "0.0.0.0:4152",
-		BroadcastAddress:  hostname,
+		BroadcastAddress:  localIP,
```

Comment:
While you might find IP a more applicable default in your situation others may find hostname more appropriate; Because this change would not be backwards compatible, and `--broadcast-address` is already exposed to make this configurable, we are unlike to make this change.